lib/rules/utils/char-source.js

```diff
@@ -80,25 +84,23 @@ function readUnicodeSequence(reader) {
  * @returns {string} A code unit.
  */
 function readOctalSequence(reader, maxLength) {
-    const posAfterBackslash = reader.pos - 1;
-    const [octalStr] = reader.source.slice(posAfterBackslash, posAfterBackslash + maxLength).match(/^[0-7]+/u);
+    const [octalStr] = reader.read(-1, maxLength).match(/^[0-7]+/u);
 
-    reader.pos = posAfterBackslash + octalStr.length;
+    reader.pos += octalStr.length - 1;
```

Comment:
Because you refactored to include a `read()` method in `SourceReader`, it might make things clearer if you also created a `advance()` method that moves `pos` rather than augmenting the value like this. So this code would become:

```suggestion
    reader.advance(octalStr.length - 1);
```

Reply:
good idea 👍🏻

Reply:
All done in b1cf05fc910e1b4c4986a652fa99c2a325be867f.